web/packages/teleport/src/Console/DocumentKubeExec/DocumentKubeExec.tsx

```diff
@@ -63,7 +63,13 @@ export default function DocumentKubeExec({ doc, visible }: Props) {
           <Indicator />
         </Box>
       )}
-      {mfa.requested && <AuthnDialog mfa={mfa} onCancel={closeDocument} />}
+      <AuthnDialog
+        {...mfa}
+        resetAttempt={() => {
+          mfa.resetAttempt();
+          closeDocument();
+        }}
+      />
```

Comment:
I think `AuthnDialog` should take the value of `mfa` as a single prop, instead of as its only props. That way, you could add an optional `onResetAttempt` handler, that this could then use to `closeDocument`, instead of having to override the `resetAttempt` method and still call through to `mfa.resetAttempt`